### PR TITLE
Build with Parameters: Permit yardmaster to build jobs that have parameters attached to them.

### PR DESCRIPTION
**Pull request by farmdawgnation**
This PR implements the ability to tell your hubot to build a job with certain parameters by merely adding those parameters onto the end of the request in chat:

```
@Hubot build my-job PARAM1=value PARAM2=value2
```

### CODE REVIEW

**Comment by farmdawgnation** [comment]
This also nukes a bunch of whitespace that @riveramj left behind.

Also what's the deal with the travis stuff? o.O

**Comment by riveramj** [comment]
the travis thing was something that i started but never got a chance to finish. i need to get on it again. its basically the `write some tests and fix this` reminder

**Comment by farmdawgnation** [comment]
Well I disabled it for branches without a `.travis.yml` heh

**Comment by riveramj** [comment]
cool thanks

File: src/yardmaster.coffee [hunk 1]
```diff
@@ -106,31 +106,48 @@ doesJobExist = (robot, msg, job, callback) ->
...
+      rawParameters = msg.match[3].trim().split(' ')
+
+      parameters = for rawParameter in rawParameters
+        [_, rawKey, rawValue] = rawParameter.trim().match(keyValueRegex)
+        encodedValue = encodeURIComponent(rawValue)
+        "#{rawKey}=#{encodedValue}"
+
+      joinedParameters = parameters.join('&')
+
+      console.log "job/#{job}/buildWithParameters?#{joinedParameters}"
```
Comment (riveramj) [inline src/yardmaster.coffee]: are you looking to have this post to the user or logs?

**Comment by riveramj** [comment]
code reads fine. Can you update the help / readme with the new info? both are out of date, and i'm going to update them, but we might as well add it now.

File: src/yardmaster.coffee (hunk 1 above)
Comment (farmdawgnation) [inline src/yardmaster.coffee]: This was debug.

**Comment by farmdawgnation** [comment]
Finally getting around to this. Sorry for the delay. :sweat:

**Comment by riveramj** [comment]
aight this looks good. thanks!

**Pull request merged**